core-v1/src/main/scala/dsl/ConstOutputType.scala

```diff
@@ -15,16 +15,20 @@ import scala.annotation.implicitNotFound
   * @tparam O the original unwrapped input type
   * @return the appropriately wrapped output scalar or functor type
   */
-@implicitNotFound("""
-Cannot find the appropriate output type when attempting to wrap a value of type ${I} into a constant.
+@implicitNotFound(
+  """Cannot find the appropriate ConstOutputType when attempting to wrap a value of type ${O}.
+     
+DSL wrapper type: ${W}
 
-Typically, this means that you are attempting to call .const on a non-Functor higher-kinded type.""")
-trait ConstOutputType[W[+_], O] {
+Typically, this means that you are calling .const on an empty collection (with an element type of Nothing).
+You can fix this by explicitly annotating the collection type parameter to something other than Nothing.""",
+)
+trait ConstOutputType[W[+_], -O] {
   type Out
 
   def wrapConst(value: O): Out
 }
 
 object ConstOutputType {
-  type Aux[W[+_], O, WO] = ConstOutputType[W, O] { type Out = WO }
+  type Aux[W[+_], -O, WO] = ConstOutputType[W, O] { type Out = WO }
```

Comment:
I found this to be helpful when determining the return type of the `.const` operator when working with `Some[A]`, as an implicit for `Option[A]` will be more specific (according to the rules of contravariance) than `Some[A]`, so it allow (and prefer) a `Traverse[Option]` over a `Traverse[Some]` (which does not exist)

Reply:
Also, I should move this comment about variance to the ScalaDocs